src/handlers/auth_handlers.rs

```diff
@@ -188,6 +196,30 @@ pub async fn login_handler(
     })
 }
 
+fn generate_secure_token() -> (String, String) {
```

Comment:
I think it will be better if you create a new file in `/utils` for  functions related to `hmac`.
This file should only contain routes.

Reply:
Ohh okay, 
Is the overall functionality of the code, I written is good ?

Reply:
It seems fine!!. I have got to test it. If you happen to test the route, provide us with details